bigbluebutton-html5/imports/ui/components/captions/pad/component.jsx

```diff
@@ -21,77 +21,247 @@ const intlMessages = defineMessages({
     id: 'app.captions.pad.ownership',
     description: 'Label for taking ownership of closed captions pad',
   },
+  dictationStart: {
+    id: 'app.captions.pad.dictationStart',
+    description: 'Label for starting speech recognition',
+  },
+  dictationStop: {
+    id: 'app.captions.pad.dictationStop',
+    description: 'Label for stoping speech recognition',
+  },
+  dictationOnDesc: {
+    id: 'app.captions.pad.dictationOnDesc',
+    description: 'Aria description for button that turns on speech recognition',
+  },
+  dictationOffDesc: {
+    id: 'app.captions.pad.dictationOffDesc',
+    description: 'Aria description for button that turns off speech recognition',
+  },
 });
 
 const propTypes = {
   locale: PropTypes.string.isRequired,
   ownerId: PropTypes.string.isRequired,
+  currentUserId: PropTypes.string.isRequired,
   padId: PropTypes.string.isRequired,
   readOnlyPadId: PropTypes.string.isRequired,
   name: PropTypes.string.isRequired,
   amIModerator: PropTypes.bool.isRequired,
+  handleAppendText: PropTypes.func.isRequired,
+  initVoiceRecognition: PropTypes.func.isRequired,
   intl: PropTypes.shape({
     formatMessage: PropTypes.func.isRequired,
   }).isRequired,
 };
 
-const Pad = (props) => {
-  const {
-    locale,
-    intl,
-    padId,
-    readOnlyPadId,
-    ownerId,
-    name,
-    amIModerator,
-  } = props;
-
-  if (!amIModerator) {
-    Session.set('openPanel', 'userlist');
+const CAPTIONS_CONFIG = Meteor.settings.public.captions;
+
+class Pad extends Component {
+  static getDerivedStateFromProps(nextProps) {
+    if (nextProps.ownerId !== nextProps.currentUserId) {
+      return ({ listening: false });
+    }
     return null;
   }
 
-  const url = PadService.getPadURL(padId, readOnlyPadId, ownerId);
-
-  return (
-    <div className={styles.pad}>
-      <header className={styles.header}>
-        <div className={styles.title}>
-          <Button
-            onClick={() => { Session.set('openPanel', 'userlist'); }}
-            aria-label={intl.formatMessage(intlMessages.hide)}
-            label={name}
-            icon="left_arrow"
-            className={styles.hideBtn}
-          />
-        </div>
-        {CaptionsService.canIOwnThisPad(ownerId)
-          ? (
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      listening: false,
+      text: '',
+    };
+
+    const { initVoiceRecognition } = props;
+    this.recognition = initVoiceRecognition();
+
+    this.toggleListen = this.toggleListen.bind(this);
+    this.handleListen = this.handleListen.bind(this);
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
```

Comment:
You can change this component to a `PureComponent` because it does this verification.